File: robocode-tankroyale-bot-api-csharp/events/BulletHitWallEvent.cs
namespace Robocode.TankRoyale
{
  /// <summary>
  /// Event occurring when a bullet has hit a wall.
  /// </summary>
  public sealed class BulletHitWallEvent : Event
  {
    /// <summary>Bullet that has hit a wall.<summary>
    BulletState Bullet { get; }

    /// <summary>
    /// Constructor.
    /// </summary>
    /// <param name="turnNumber">Turn number.</param>
    /// <param name="bullet">Bullet that has hit a wall.</param>
    /// <returns></returns>
    public BulletHitWallEvent(int turnNumber, BulletState bullet) : base(turnNumber) => Bullet = bullet;
  }
}

File: bot-api/typescript/src/events/BulletHitWallEvent.ts
import { BotEvent } from "./BotEvent.js";
import { BulletState } from "../BulletState.js";

/** Event occurring when a bullet has hit a wall. */
export class BulletHitWallEvent extends BotEvent {
  readonly bullet: BulletState;

  constructor(turnNumber: number, bullet: BulletState) {
    super(turnNumber);
    this.bullet = bullet;
  }
}
